src/JWT/Algorithms/HMACSHAAlgorithmFactory.cs

```diff
@@ -5,16 +5,28 @@ namespace JWT.Algorithms
     /// <inheritdoc />
     public class HMACSHAAlgorithmFactory : JwtAlgorithmFactory
     {
+        private readonly byte[] _key;
+
+        public HMACSHAAlgorithmFactory()

```

Comment:
Why do all classes need an empty ctor? Both the algos and the factories?

Reply:
I didn't want to introduce breaking changes for the user code like:
```cs
JwtBuilder.Create().WithAlgorithm(...)...;
JwtBuilder.Create().WithAlgorithm(...).WithSecret(...).Encode(...);
JwtBuilder.Create().WithAlgorithmFactory(...)...;
```
I can remove empty ctors if you want to